flycheck.el

```diff
@@ -6275,11 +6275,30 @@ See URL `http://www.haskell.org/ghc/'."
   :modes (haskell-mode literate-haskell-mode)
   :next-checkers ((warning . haskell-hlint)))
 
+(flycheck-def-config-file-var flycheck-hlintrc haskell-hlint "HLint.hs"
+  :safe #'stringp)
+
+(flycheck-def-option-var flycheck-hlint-options
```

Comment:
Please rename this option to `flycheck-hlint-language-extensions`, and change it so that users don't have to specify `-X`.